protocols/kad/src/behaviour.rs

```diff
@@ -2457,6 +2492,17 @@ where
             self.put_record_job = Some(job);
         }
 
+        // Poll bootstrap periodically.
+        if let Some(mut bootstrap_timer) = self.bootstrap_timer.take() {
+            if let Poll::Ready(()) = Pin::new(&mut bootstrap_timer).poll(cx) {
+                if let Some(interval) = self.bootstrap_interval {
+                    bootstrap_timer.reset(interval);
+                    let _ = self.bootstrap();
```

Comment:
I'd rename the `on_new_kademlia_node` function and call it here as well. It is the same piece of logic that we want to execute here (i.e. don't do another bootstrap if are currently running one).

Reply:
Renamed to handle_bootstrap for now, if any better idea please let me know :)